pkg/resources/resources.go

```diff
@@ -55,6 +57,7 @@ type structured struct {
 type Client struct {
 	ctx        context.Context
 	dynamic    dynamic.Interface
+	mapper     *restmapper.DeferredDiscoveryRESTMapper
```

Comment:
one TODO is to figure out how to handle the mapper in tests 🤔 

maybe adding an local interface since there doesn't seem to be one for it in `restmapper`? i am open to suggestions if anyone has any other ideas on how to structure this for tests

Reply:
Not sure If I understand the question. I'm guessing you mean providing a mock, right? Why not using the [RestMapper](https://pkg.go.dev/k8s.io/apimachinery@v0.24.2/pkg/api/meta#RESTMapper) interface? Am I missing something?

Reply:
ah, nice. thanks! that's the interface i was looking for